packages/dds/task-manager/src/taskManager.ts

```diff
@@ -478,10 +530,21 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         this.connectionWatcher.on("disconnect", disconnectHandler);
         this.completedWatcher.on("completed", checkIfCompleted);
 
-        if (!this.connected) {
+        if (!this.isAttached()) {
+            // Simulate auto-ack in detached scenario
+            assert(this.clientId !== undefined, "clientId is undefined");
+            this.onVolunteerAck(taskId, this.clientId);
+            // If we volunteered using placeHolderClientId, then we need to watch for when we attach and are assigned a
+            // real clientId. At that point we should re-enter the queue with a real volunteer op.
+            if (this.clientId === this.placeholderClientId) {
+                this.attachedWatcher.once("attached", () => {
```

Comment:
Maybe just use connectionWatcher "connect" here rather than explicitly watching for "attached"?  Connection is when we get assigned the real clientId and I think this maps better to the semantics of what we need to do ("you lose the task when you start the attach, and after that it's the normal case where you can only volunteer when connected")